### PR TITLE
Bigtable: New 'retry' parameter for 'Table.read_row()'

### LINKED ISSUE #7955: bigtable: Implement consistent retry mechanisms
**Is your feature request related to a problem? Please describe.**
It is frustrating to utilize a provided retry mechanism for one API call, and being forced to implement our own retry mechanism for a very similar API call because retries aren't provided.

https://googleapis.github.io/google-cloud-python/latest/bigtable/table.html#google.cloud.bigtable.table.Table.read_row

https://googleapis.github.io/google-cloud-python/latest/bigtable/table.html#google.cloud.bigtable.table.Table.read_rows

 **Describe the solution you'd like**
`table.Table.read_rows` accepts a Retry, but `table.Table.read_row` does not. It would be nice if `table.Table.read_row` accepted the same retry mechanism.
 
**Describe alternatives you've considered**
In the absence of a retry mechanism built into read_row, we have written our own.

### PR DESCRIPTION
**Pull request by mf2199**
Fixes #7955 . Reformatted.

### CODE REVIEW

File: bigtable/tests/unit/test_table.py [hunk 1]
```diff
@@ -627,6 +627,33 @@ def test_mutate_rows(self):
...
+            deadline=120.0,  # 2 minutes
+        )
+        table.read_row(self.ROW_KEY, retry=custom_retry)
+        args, kwargs = table.read_rows.call_args
+        self.assertEqual(kwargs["retry"], custom_retry)
+
+        table.read_row(self.ROW_KEY)
+        args, kwargs = table.read_rows.call_args
+        self.assertEqual(kwargs["retry"], row_data.DEFAULT_RETRY_READ_ROWS)
+
```
Comment (tseaver) [inline bigtable/tests/unit/test_table.py]: We generally prefer to call the function-under-test once per testcase.  Can you please split out this last call into a separate `test_read_row_default_retry` testcase?  E.g.:

```python
    def test_read_row_default_retry(self):
        from google.cloud.bigtable import row_data

        credentials = _make_credentials()
        client = self._make_client(project="project-id", credentials=credentials)
        instance = client.instance(instance_id=self.INSTANCE_ID)
        table = self._make_one(self.TABLE_ID, instance)
        table.read_rows = mock.Mock(return_value=[])

        table.read_row(self.ROW_KEY)
        args, kwargs = table.read_rows.call_args
        self.assertEqual(kwargs["retry"], row_data.DEFAULT_RETRY_READ_ROWS)
```

Comment (mf2199) [reply bigtable/tests/unit/test_table.py]: Done, and noted for the future, thanks. Some extra blank lines have been added in `noxfile.py` as per PEP8 style guide.

**Comment by mf2199** [comment]
There are multiple warning while running the unit tests, apparently due to the deprecation of the `channel` argument:

`============================== warnings summary ===============================
tests/unit/test_app_profile.py::TestAppProfile::test_create_routing_any
  C:\git\google-cloud-python-2\bigtable\tests\unit\test_app_profile.py:400: PendingDeprecationWarning: The `channel` argument is deprecated; use `transport` instead.
    channel=channel

tests/unit/test_app_profile.py::TestAppProfile::test_create_routing_single
  C:\git\google-cloud-python-2\bigtable\tests\unit\test_app_profile.py:453: PendingDeprecationWarning: The `channel` argument is deprecated; use `transport` instead.
    channel=channel

tests/unit/test_app_profile.py::TestAppProfile::test_update_app_profile_routing_any
  C:\git\google-cloud-python-2\bigtable\tests\unit\test_app_profile.py:523: PendingDeprecationWarning: The `channel` argument is deprecated; use `transport` instead.
    channel=channel

tests/unit/test_app_profile.py::TestAppProfile::test_update_app_profile_routing_single
  C:\git\google-cloud-python-2\bigtable\tests\unit\test_app_profile.py:582: PendingDeprecationWarning: The `channel` argument is deprecated; use `transport` instead.
    channel=channel

tests/unit/test_cluster.py::TestCluster::test_create
  C:\git\google-cloud-python-2\bigtable\tests\unit\test_cluster.py:378: PendingDeprecationWarning: The `channel` argument is deprecated; use `transport` instead.
    channel=channel

tests/unit/test_cluster.py::TestCluster::test_update
  C:\git\google-cloud-python-2\bigtable\tests\unit\test_cluster.py:437: PendingDeprecationWarning: The `channel` argument is deprecated; use `transport` instead.
    channel=channel

tests/unit/test_row_data.py::TestPartialRowsData::test_multiple_chunks
  C:\git\google-cloud-python-2\bigtable\tests\unit\test_row_data.py:505: PendingDeprecationWarning: The `channel` argument is deprecated; use `transport` instead.
    data_api = bigtable_client.BigtableClient(channel=channel)

tests/unit/test_row_data.py::TestPartialRowsData::test_state_new_row_w_row
  C:\git\google-cloud-python-2\bigtable\tests\unit\test_row_data.py:465: PendingDeprecationWarning: The `channel` argument is deprecated; use `transport` instead.
    data_api = bigtable_client.BigtableClient(channel=channel)

-- Docs: https://docs.pytest.org/en/latest/warnings.html
`

Since the `channel` is deprecated in the favour of the `transport`, and these do not seem to test the `channel` as such, maybe it's a good idea to refactor the tests as well?

**Comment by tseaver** [comment]
The most straightforward way to prevent those warnings is to wrap each of those channels in a transport and pass *that* to the cilent.  My memory is that some of the tests assume / patch some very low-level stuff about the channel, but I could be wrong.

**Comment by tseaver** [comment]
Overall design of retries for manual clients is tracked in #9298.

**Pull request closed without merging**